text/0000-decorators.md

```diff
@@ -0,0 +1,1230 @@
+- Start Date: 2018-10-20
+- RFC PR: (leave this empty)
+- Ember Issue: (leave this empty)
+
+# Decorators
+
+## Summary
+
+Native classes are now officially supported in Ember, but currently their usage
+is very limited. Core Ember features such as computed properties, actions, and
+service injections have no publicly supported APIs to be used with native class
+syntax.
+
+Decorators provide a way to abstract functionality and improve the developer
+experience of working with native classes.  This RFC outlines the implementation
```

Comment:
Decorators provide a mechanism for expositing these features to ES6 consumers, while ensuring interoperability with code written in the traditional `Ember.Object.extend` way